Avoid duplicate "warning:" or "error:" output

tautschnig opened the pull request:
The gcc message handler already took care of this; make the console message
handler do the same.

kroening commented:
Uhoh, lots of regressions check for "CONVERSION ERROR". Not clear whether to change the tests or do a special case for that.

tautschnig commented:
Assigning to myself for further work at a later time. No urgency/dependency here for anyone.

tautschnig committed:
Avoid duplicate "warning:" or "error:" prefixes in the C/C++ front-end

The gcc and cl message handlers already take care of this, and we had a
mix of some messages including the "warning:" or "error:" prefix when
others don't.

tautschnig committed:
goto-cc modes: interpret command-line options to set default verbosity to warning

We already did this for GCC, but we can also do this for other modes
(ARMCC, CodeWarrior, LD, CL, LINK).

tautschnig committed:
GCC and CL message handler: print warnings as errors

We already treated warnings as errors (if requested via command-line
options), but we should also mimic the output presented by CL/GCC with
those command-line options set.

tautschnig commented:
> Uhoh, lots of regressions check for "CONVERSION ERROR". Not clear whether to change the tests or do a special case for that.

I have revamped this PR, and none of the "CONVERSION ERROR" messages should be affected. It's really just to avoid "warning: warning: " being printed to users.

martin-cs reviewed (approved):
Much respect for resolving a 2017 issue!

NlightNFotis commented:
Hi @tautschnig, would it be possible to rebase this on top of current `develop` so that the CI checks get updated?

tautschnig commented:
> Hi @tautschnig, would it be possible to rebase this on top of current `develop` so that the CI checks get updated?

Done!

NlightNFotis commented:
Hi @tautschnig, could you push this one again please?

It seems one of the CI checks has failed to initialise, blocking the whole PR from getting merged despite approvals.

TGWDB commented:
@tautschnig Looks like this would probably pass if it was pushed with a tiny change to rerun CI jobs?

TGWDB commented:
Closing due to age (no further comment on PR content), please reopen with rebase on develop if you intent to continue this work.

tautschnig commented:
Re-opening: this is just waiting for a review.

TGWDB commented:
@remi-delmas-3000 @kroening Woulda review be possible?

kroening commented on regression/ansi-c/linking_conflicts1/test.desc:
```diff
@@ -3,7 +3,7 @@ main.c
-error: conflicting function declarations 'bar2'
+conflicting function declarations 'bar'
+conflicting function declarations 'bar2'
 --
```
Make that $error: conflicting...., so it actually tests that the error isn't there twice.

tautschnig replied:
Amended.

kroening commented on regression/cbmc/incomplete-structs/test.desc:
```diff
@@ -1,8 +1,8 @@
-warning: pointer parameter types differ between declaration and definition "bar"
-warning: pointer parameter types differ between declaration and definition "foo"
+pointer parameter types differ between declaration and definition 'bar'
+pointer parameter types differ between declaration and definition 'foo'
```
same here

tautschnig replied:
Amended.

kroening commented on src/goto-cc/cl_message_handler.h:
```diff
@@ -29,6 +29,16 @@ class cl_message_handlert : public console_message_handlert
+
+  /// With \p yes set to \c true, prefix warnings with an error message.
+  /// \param yes: Whether or not to prefix warnings.
+  void print_warnings_as_errors(bool yes)
```
Do we really need that?

tautschnig replied:
I'd say it depends on how closely we'd like to mimic the behaviour of Visual Studio and GCC. Prior to this change, we just printed "warning" but nonetheless failed with a non-zero exit code when `Werror` was used. With this change, we would actually print "error" for such now-fatal warnings, as those compilers do.

kroening replied:
Feels a bit cosmetic, but why not.
